plugins/outputs/circonus/check.go

```diff
@@ -0,0 +1,332 @@
+		filterCriteria *apiclient.SearchFilterType) (*[]apiclient.CheckBundle, error)
+	CreateCheckBundle(cfg *apiclient.CheckBundle) (*apiclient.CheckBundle, error)
+}
+
+// getAPIClient creates a Circonus API client.
+func (c *Circonus) getAPIClient() error {
+	apiCfg := &apiclient.Config{
+		TokenKey: c.APIToken,
+		TokenApp: c.APIApp,
+		URL:      c.APIURL,
+	}
+
+	if c.APITLSCA != "" {
+		cert, err := ioutil.ReadFile(c.APITLSCA)
+		if err != nil {
+			return fmt.Errorf("unable to configure Circonus API client: %w", err)
+		}
+
+		cp := x509.NewCertPool()
+		if !cp.AppendCertsFromPEM(cert) {
+			return fmt.Errorf("unable to add Circonus API CA certificate to certtificate pool")
+		}
+
+		apiCfg.TLSConfig = &tls.Config{RootCAs: cp}
+	}
+
+	if c.APIInsecureSkipVerify {
+		if apiCfg.TLSConfig == nil {
+			apiCfg.TLSConfig = &tls.Config{InsecureSkipVerify: true}
+		} else {
+			apiCfg.TLSConfig.InsecureSkipVerify = true
+		}
+	}
+
+	cli, err := apiclient.New(apiCfg)
+	if err != nil {
+		return fmt.Errorf("unable to create Circonus API client: %w", err)
+	}
+
+	c.api = cli
+
+	return nil
+}
+
+// getCheckBroker returns the broker CID for a Circonus check.
+func (c *Circonus) getCheckBroker(name string) (string, error) {
+	if c.api == nil {
+		return "", fmt.Errorf("unable to get check broker: Circonus API not initialized")
+	}
+
+	search := apiclient.SearchQueryType("(active:1)(type:httptrap)(name:" + name + ")")
+	checks, err := c.api.SearchCheckBundles(&search, nil)
+	if err != nil {
+		return "", fmt.Errorf("unable to search check bundles: %w", err)
+	}
+
+	if checks == nil {
+		return "", nil
+	}
+
+	if len(*checks) == 0 {
+		return "", nil
+	}
+
+	if len((*checks)[0].Brokers) == 0 {
+		return "", nil
+	}
+
+	return (*checks)[0].Brokers[0], nil
+}
+
+// getRandomBroker returns the CID and CN of a random eligible broker.
+func (c *Circonus) getRandomBroker() (string, string, error) {
+	if c.api == nil {
+		return "", "", fmt.Errorf("unable to get random broker CID: Circonus API not initialized")
+	}
+
+	brokers, err := c.api.FetchBrokers()
+	if err != nil {
+		return "", "", fmt.Errorf("unable to fetch brokers: %w", err)
+	}
+
+	if brokers == nil {
+		return "", "", nil
+	}
+
+	if len(*brokers) == 0 {
+		return "", "", nil
+	}
+
+	brokerCID := ""
+	brokerCN := ""
+	for _, broker := range *brokers {
+		if len(broker.Details) == 0 {
+			continue
+		}
+
+		if broker.Details[0].IP == nil || *broker.Details[0].IP == "" {
+			continue
+		}
+
+		exclude := false
+		for _, excid := range c.ExcludeBrokers {
+			if broker.CID == excid {
+				exclude = true
+				break
+			}
+		}
+
+		if exclude {
+			continue
+		}
+
+		httpTrap := false
+
+		for _, mod := range broker.Details[0].Modules {
+			if mod == "httptrap" {
+				httpTrap = true
+				break
+			}
+		}
+
+		if !httpTrap {
+			continue
+		}
+
+		brokerCID = broker.CID
+		brokerCN = broker.Details[0].CN
+		break
+	}
+
+	return brokerCID, brokerCN, nil
```

Comment:
Wouldn't
```suggestion
		return broker.CID, broker.Details[0].CN, nil
	}

	return "", "", nil
```
be more clear?